tox.ini

```diff
@@ -129,6 +129,12 @@ description = static type checking with pyright
 deps = pyright==1.1.338
 commands = pyright src/snowflake/snowpark/_internal/analyzer
 
+[testenv:mypy]
```

Comment:
In snow pandas, we actually add mypy into precommit check, see we need to change .pre-commit-config.yaml like this 
https://github.com/snowflakedb/snowpandas/blob/6f12d96cfb3db7fe2e405d0dfae0b9f6af60ebeb/.pre-commit-config.yaml#L89

so when we develop locally and run `pre-commit`, this check . 

In this way, if we run precommit locally or in CI, it will run mypy too. I would suggest doing the same thing. But we can do in a separate PR.

Reply:
```
(snowpandas) ➜  snowpark-python git:(mypy-support mypy --config-file mypy.ini \
                src/snowflake/snowpark/_internal
src/snowflake/snowpark/_internal/udf_utils.py:705: error: Argument 1 to "generate_source_code" has incompatible type "Callable[..., Any]"; expected "Union[FunctionType, BuiltinFunctionType]"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py:564: error: Missing return statement  [return]
src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py:697: error: Missing positional argument "child" in call to "pivot_statement"  [call-arg]
src/snowflake/snowpark/_internal/analyzer/analyzer.py:984: error: Missing positional argument "source_plan" in call to "pivot" of "SnowflakePlanBuilder"  [call-arg]
src/snowflake/snowpark/_internal/analyzer/analyzer.py:990: error: Item "LogicalPlan" of "Union[List[Expression], LogicalPlan, None]" has no attribute "__iter__" (not iterable)  [union-attr]
src/snowflake/snowpark/_internal/analyzer/analyzer.py:990: error: Item "None" of "Union[List[Expression], LogicalPlan, None]" has no attribute "__iter__" (not iterable)  [union-attr]
src/snowflake/snowpark/_internal/analyzer/analyzer.py:995: error: Argument 4 to "pivot" of "SnowflakePlanBuilder" has incompatible type "SnowflakePlan"; expected "Optional[str]"  [arg-type]
src/snowflake/snowpark/_internal/analyzer/analyzer.py:996: error: Argument 5 to "pivot" of "SnowflakePlanBuilder" has incompatible type "Pivot"; expected "SnowflakePlan"  [arg-type]
Found 8 errors in 3 files (checked 27 source files)

```
my local test failed.